jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -0,0 +1,156 @@
+/*[INCLUDE-IF CRIU_SUPPORT]*/
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.eclipse.openj9.criu;
+
+import java.nio.file.Path;
+
+/**
+ * CRIU Support API
+ */
+public final class CRIUSupport {
+	
+	/**
+	 * CRIU operation return type
+	 */
+	public static enum CRIUResultType {
+		/**
+		 * The operation was successful.
+		 */
+		SUCCESS,
+		/**
+		 * The requested operation is unsupported.
+		 */
+		UNSUPPORTED_OPERATION,
+		/**
+		 * The arguments provided for the operation are invalid.
+		 */
+		INVALID_ARGUMENTS,
+		/**
+		 * A system failure occurred when attempting to checkpoint the JVM.
+		 */
+		SYSTEM_CHECKPOINT_FAILURE,
+		/**
+		 * A JVM failure occurred when attempting to checkpoint the JVM.
+		 */
+		JVM_CHECKPOINT_FAILURE,
+		/**
+		 * A non-fatal JVM failure occurred when attempting to restore the JVM.
+		 */
+		JVM_RESTORE_FAILURE;
+	}
+	
+	/**
+	 * CRIU operation result. If there is a system failure the system error code will be set.
+	 * If there is a JVM failure the throwable will be set.
+	 *
+	 */
+	public static class CRIUResult {
+		private CRIUResultType type;
+
+		private Throwable throwable;
```

Comment:
The class and its fields can be `final`.

Reply:
You fixed the fields, why not the class?

Reply:
updated.